src/extension.ts

```diff
@@ -271,31 +271,71 @@ class TabNine {
       "version": version,
       "request": any_request
     };
+
+    const unregisterFunctions = [];
+
     const request = JSON.stringify(any_request) + '\n';
-    return new Promise<any>((resolve, reject) => {
+    let response = new Promise<any>((resolve, reject) => {
       try {
         if (!this.isChildAlive()) {
           this.restartChild();
         }
         if (!this.isChildAlive()) {
           reject(new Error("TabNine process is dead."))
         }
-        this.rl.once('line', (response) => {
+        const onResponse: (input: any) => void = (response) => {
           let any_response: any = JSON.parse(response.toString());
           resolve(any_response);
-        });
+        };
+        this.rl.once('line', onResponse);
+
+        unregisterFunctions.push(() => this.rl.removeListener('line', onResponse));
```

Comment:
do we need to unregister events that were register with `once` ?

Reply:
Yes, if the promise was resolved or rejected, and these were not fired yet, we need to unregister or we are going to leak.
Specifically, the process does not exit under normal circumstances, so if we just register for it with evey request, we are going to leak

Reply:
👍

Reply:
nice catch...